composeApp/src/main/kotlin/io/github/couchtracker/tmdb/TmdbRating.kt

```diff
@@ -1,28 +1,40 @@
 package io.github.couchtracker.tmdb
 
 import app.moviebase.tmdb.model.TmdbRatingItem
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import java.text.NumberFormat
 
-data class TmdbRating(
+data class TmdbRating private constructor(
     val average: Float,
     val count: Int,
+    val formatted: String,
 ) {
     init {
         require(count > 0)
     }
 
-    // TODO put in composable function
-    fun format() = NumberFormat.getInstance().apply {
-        minimumFractionDigits = 1
-        maximumFractionDigits = 1
-    }.format(average) + "★"
+    companion object {
+        suspend fun of(average: Float, count: Int): TmdbRating {
+            return withContext(Dispatchers.Default) {
+                TmdbRating(
+                    average = average,
+                    count = count,
+                    formatted = NumberFormat.getInstance().apply {
```

Comment:
It's OK for now, but this is only marginally better than the previous iteration. If the locale changes, this won't be recomputed.